iphone/CoreApi/CoreApi/Logger/Logger.mm

```diff
@@ -132,10 +115,54 @@ bool AssertMessage(base::SrcPoint const & src, std::string const & message)
 
 // MARK: - Private
 
-+ (BOOL)isFileLoggingEnabled {
-  return [self logger].fileHandle != nil;
++ (void)enableFileLogging {
+  Logger * logger = [self logger];
+  NSFileManager * fileManager = [NSFileManager defaultManager];
+
+  // Create a log file if it doesn't exist and setup file handle for writing.
+  NSString * filePath = [[fileManager temporaryDirectory] URLByAppendingPathComponent:kLogFileName].path;
+  if (![fileManager fileExistsAtPath:filePath]) {
+    [fileManager createFileAtPath:filePath contents:nil attributes:nil];
+  }
+  NSFileHandle * fileHandle = [NSFileHandle fileHandleForWritingAtPath:filePath];
+  if (fileHandle == nil) {
+    LOG(LERROR, ("Failed to open log file for writing"));
+    [self setFileLoggingEnabled:NO];
+    return;
+  }
+  // Clean up the file if it exceeds the maximum size.
```

Comment:
What is the case when a log file is not deleted here?

Reply:
The absence of a file is checked up some lines above.
Here we do not delete the file but only clean the content.
The same approach is in Android.

Reply:
1. Isn't the file deleted when user disables logging?
2. When user enables logging, the previous log file should not be present, right?

Reply:
1. when user disables logging the file will be deleted
2. yes!